internal/db/kvs/pogreb/pogreb.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2019-2024 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pogreb
+
+import (
+	"context"
+	"os"
+	"reflect"
+
+	"github.com/akrylysov/pogreb"
+
+	"github.com/vdaas/vald/internal/conv"
+	"github.com/vdaas/vald/internal/errors"
+)
+
+// Pogreb represents an interface for operating the pogreb database.
+type Pogreb interface {
+	Set(key string, val []byte) error
+	Get(key string) ([]byte, bool, error)
+	Delete(key string) error
+	Range(ctx context.Context, f func(key string, val []byte) bool) error
+	Len() uint32
+	Close(remove bool) error
+}
+
+type db struct {
+	db   *pogreb.DB
+	opts *pogreb.Options
+	path string
+}
+
+// New returns a new pogreb instance.
+// If the directory path does not exist, it creates a directory for database.
+// If opts is nil, it uses default options.
+func New(opts ...Option) (_ Pogreb, err error) {
+	db := new(db)
+	for _, opt := range append(deafultOpts, opts...) {
+		if err := opt(db); err != nil {
+			return nil, errors.ErrOptionFailed(err, reflect.ValueOf(opt))
+		}
+	}
+
+	// If db.opts is nil, an default value is used.
```

Comment:
```suggestion
	// If db.opts is nil, a default value is used.
```

Reply:
Thanks for catching it :pray: I will apply your suggestion!